Composer/packages/lib/indexers/src/validations/expressionValidation/index.ts

```diff
@@ -49,8 +49,10 @@ export const validateExpressions: ValidateFunc = (
     let errorMessage = '';
     let warningMessage = '';
     try {
-      newCache[value] = cache?.[value] ? cache[value] : checkExpression(value, required, types);
-      errorMessage = checkReturnType(newCache[value], types);
+      //the cacheKey adds the type of the value at the end, this allows a new indexing system to prevent the overwriting in cache
```

Comment:
```suggestion
      // The cacheKey adds the type of the value at the end, this allows a new indexing system to prevent the overwriting in cache.
```